File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/schema/relationshiptype/RelationshipTypeName.ts
module api.schema.relationshiptype {

    import ApplicationKey = api.application.ApplicationKey;
    import ApplicationBasedName = api.application.ApplicationBasedName;

    export class RelationshipTypeName extends api.application.ApplicationBasedName {

        static REFERENCE: RelationshipTypeName = new RelationshipTypeName(
            `${ApplicationKey.SYSTEM}${ApplicationBasedName.SEPARATOR}reference`
        );

        constructor(name: string) {
            api.util.assertNotNull(name, "RelationshipType name can't be null");
            let parts = name.split(api.application.ApplicationBasedName.SEPARATOR);
            super(ApplicationKey.fromString(parts[0]), parts[1]);
        }

    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/schema/relationship/RelationshipTypeName.java
package com.enonic.xp.schema.relationship;

import com.enonic.xp.annotation.PublicApi;
import com.enonic.xp.app.ApplicationKey;
import com.enonic.xp.schema.BaseSchemaName;

@PublicApi
public final class RelationshipTypeName
    extends BaseSchemaName
{
    public static final RelationshipTypeName REFERENCE = new RelationshipTypeName( ApplicationKey.SYSTEM, "reference" );

    public static final RelationshipTypeName PARENT = new RelationshipTypeName( ApplicationKey.SYSTEM, "parent" );

    private RelationshipTypeName( final String name )
    {
        super( name );
    }

    private RelationshipTypeName( final ApplicationKey applicationKey, final String localName )
    {
        super( applicationKey, localName );
    }

    public static RelationshipTypeName from( final ApplicationKey applicationKey, final String localName )
    {
        return new RelationshipTypeName( applicationKey, localName );
    }

    public static RelationshipTypeName from( final String relationshipTypeName )
    {
        return new RelationshipTypeName( relationshipTypeName );
    }
}
